### PR TITLE
Update: build (use hatch), py versions, obsolete files

### LINKED ISSUE #108: docs build is failing
https://readthedocs.org/projects/era5cli/builds/

See comments by @sverhoeven on #106

### PR DESCRIPTION
**Pull request by BSchilperoort**
- Where possible, settings have been moved to the `pyproject.toml` file
- Building the package has been moved to `hatch`. This is way faster, and can be properly configured using `pyproject.toml`
- Python versions have been updated where possible
- Obsolete files have been removed (e.g. zenodo.json, configurations now in pyproject)
- The codecov action and configuration have been updated. The version used was deprecated and not generating coverage reports anymore
- Codecov coverage generation and report has been split off into a separate action. (coverage was uploaded for every pyversion in the test matrix)
- Building the package is now added to the CI test.
- The codacy configuration file has been removed. Codacy has not been working anymore and would require setting up again. Switching to e.g. SonarCloud might be easier then.
- ReadTheDocs badge has been set to "stable".

Closes #108, #119, #128, #129

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,4 +1,78 @@
...
+    {name = "Jerom Aerts"},
+    {name = "Berend Weel"},
+    {name = "Rolf Hut"},
+    {name = "Klaus Zimmermann"},
+    {name = "Peter Kalverla"},
+    {name = "Barbara Vreede", email="b.vreede@esciencecenter.nl"},
+    {name = "Aytaç Paçal"},
+    {name = "Stef Smeets"},
+    {name = "Stefan Verhoeven"},
+    {name = "Bart Schilperoort", email = "b.schilperoort@esciencecenter.nl" }
```
Comment (bvreede) [inline pyproject.toml:33]:
```suggestion
    {name = "Elizaveta Malinina"},
    {name = "Bart Schilperoort", email = "b.schilperoort@esciencecenter.nl" }
```

File: tests/test_utils.py [hunk 2]
```diff
@@ -1,9 +1,7 @@
 """Tests for era5cli utility functios."""
```
Comment (bvreede) [inline tests/test_utils.py:1]:
```suggestion
"""Tests for era5cli utility functions."""
```
typo 🙃

**Review by bvreede — APPROVED** [review]
Awesome @BSchilperoort, thanks! Such an improvement.

One extra thing would be to double check whether the changes may have broken the conda upload workflow at https://github.com/conda-forge/era5cli-feedstock.

Thanks!

**Comment by BSchilperoort** [comment]
Thanks for the review!

> One extra thing would be to double check whether the changes may have broken the conda upload workflow at https://github.com/conda-forge/era5cli-feedstock.

It seems like updating the conda force workflow needs to be done manually for each new release anyway? 

From the conda-forge/era5cli-feedstock readme:

> If you would like to [...] build a new package version, please fork this repository and submit a PR. 
Upon submission, your changes will be run on the appropriate platforms to give the reviewer an opportunity to confirm that the changes result in a successful build.

**Comment by bvreede** [comment]
I'm not 100% sure how this works — @zklaus set this up — but as far as I'm aware this is an automatic workflow. Last new release was rapidly updated on `conda-forge` as well.

**Comment by zklaus** [comment]
Thanks for modernizing the "boring" infrastructure parts and thanks for considering conda-forge. Conda-forge is highly automized, meaning there are a few bots running, monitoring Pypi and a few other release channels. As soon as a new release drops, the bot will notice and open a PR with updated to the recipe to get the new version. If a change to the build steps is necessary (perhaps we need to move from `pip install ...` to `hatch install ...`?), this needs to be done manually, so it's good that you made me aware of these changes and I'll get back to you if I need help for the conda-forge update when the next release comes.

**Pull request merged**